### PR TITLE
E19 view following list

### PR DESCRIPTION
**Pull request by hiixdayah**
## Description:
- View Following List
- View User Feed 
## Checklist:
- [x] Merged latest develop
- [x] PR title makes sense

### CODE REVIEW

File: src/services/mentorship.service.ts [hunk 1]
```diff
@@ -543,5 +541,42 @@ export default class MentorshipService {
...
+  accountId: string 
+) {
+
+  const user = await User.findByPk(accountId); 
+  if (!user) throw new Error(ERRORS.USER_DOES_NOT_EXIST);
+  
+  if (userId !== accountId)  
+    throw new Error(
+      httpStatusCodes.getStatusText(httpStatusCodes.UNAUTHORIZED)
+    );
```
Comment (kengthong) [inline src/services/mentorship.service.ts]: Should use the requireSameUserOrAdmin middleware in the route to handle this instead. Admin should be able to get too right?

File: src/services/social.service.ts [hunk 2]
```diff
@@ -104,6 +105,18 @@ export default class SocialService {
       });
     }
 
+  public static async getUserFeed(accountId: string) {
+    const user = await User.findByPk(accountId); 
+    if (!user) throw new Error(ERRORS.USER_DOES_NOT_EXIST); 
+
+    if (user.isPrivateProfile === true) throw new Error (SOCIAL_ERRORS.PRIVATE_USER)
```
Comment (kengthong) [inline src/services/social.service.ts]: If its a private user, can the private user's follower list be able to access the private user's feed? If can then you would have to do more checks here.

File: src/services/social.service.ts [hunk 3]
```diff
@@ -199,6 +212,15 @@ public static async addUserToFollowingList(
     
         }
 
-      
+        public static async getFollowingList(accountId: string, userId: string) {
+          if (accountId !== userId)
+          throw new Error(
+        httpStatusCodes.getStatusText(httpStatusCodes.UNAUTHORIZED)
+      );
```
Comment (kengthong) [inline src/services/social.service.ts]: Can User A view User B's following list too? If can then you might have to add more checks

**Review by kengthong — CHANGES_REQUESTED** [review]

File: src/services/mentorship.service.ts [hunk 4]
```diff
@@ -516,32 +521,37 @@ export default class MentorshipService {
...
 
-  //get ONE testimonial 
-  public static async getTestimonial(
-    testimonialId: string,
-    accountId: string 
-  ) {
-    const testimonial = await Testimonial.findByPk(testimonialId, {
-      include: [MentorshipContract],
+  //get list of testimonials by mentorship listing for Sensei
+  public static async getTestimonial(mentorshipListingId: string) {
```
Comment (kengthong) [inline src/services/mentorship.service.ts]: could you refactor the name of this function? Add an s when its plural so that otehrs dont get confused

**Review by kengthong — CHANGES_REQUESTED** [review]

**Review by ianleck — DISMISSED** [review]

**Review by ianleck — APPROVED** [review]

**Pull request merged**